### PR TITLE
TRUNK-5944: LocationAttributeType Domain - Switching from Hibernate Mappings to Annotations

### PR DESCRIPTION
**Pull request by ManojLL**
## Description of what I changed

replace hibernate xml mappings with annotations and removed relevant xml file.
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5944

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Review by mherman22 — APPROVED** [review]

File: api/src/main/java/org/openmrs/LocationAttributeType.java [hunk 1]
```diff
@@ -9,16 +9,35 @@
...
  * @since 1.9
  */
+@Entity
+@Table(name = "location_attribute_type")
 public class LocationAttributeType extends BaseAttributeType<Location> implements AttributeType<Location> {
-	
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "location_attribute_type_id_seq")
+	@GenericGenerator(
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/LocationAttributeType.java:35]: Is there any reason why this is different from both `VisitAttributeType` and `ProviderAttributeType`?

Comment (ManojLL) [reply api/src/main/java/org/openmrs/LocationAttributeType.java:35]: There is no distinction between these entities. However, in the XML mapping, the following configuration for the ID generator has been employed:

`
<generator class="native">
    <param name="sequence">location_attribute_type_location_attribute_type_id_seq</param>
</generator>
`

Therefore, I opted for GenerationType.SEQUENCE to align with the specified sequence configuration in the XML mapping.

Comment (ManojLL) [reply api/src/main/java/org/openmrs/LocationAttributeType.java:35]: @dkayiwa

Comment (ManojLL) [reply api/src/main/java/org/openmrs/LocationAttributeType.java:35]: @dkayiwa any update on this ?

**Pull request merged**